WooCommerce/Classes/ViewRelated/Coupons/CouponListViewController.swift

```diff
@@ -33,50 +33,18 @@ final class CouponListViewController: UIViewController, GhostableViewController
     ///
     private lazy var footerEmptyView = UIView(frame: .zero)
 
-    /// Create a `UIBarButtonItem` to be used as the search button on the top-left.
-    ///
-    private lazy var searchBarButtonItem: UIBarButtonItem = {
-        let button = UIBarButtonItem(image: .searchBarButtonItemImage,
-                                     style: .plain,
-                                     target: self,
-                                     action: #selector(displaySearchCoupons))
-        button.accessibilityTraits = .button
-        button.accessibilityLabel = Localization.accessibilityLabelSearchCoupons
-        button.accessibilityHint = Localization.accessibilityHintSearchCoupons
-        button.accessibilityIdentifier = "coupon-search-button"
-
-        return button
-    }()
-
-    /// Create a `UIBarButtonItem` to be used as the create coupon button on the top-right.
-    ///
-    private lazy var createCouponButtonItem: UIBarButtonItem = {
-        let button = UIBarButtonItem(image: .plusImage,
-                style: .plain,
-                target: self,
-                action: #selector(displayCouponTypeBottomSheet))
-        button.accessibilityTraits = .button
-        button.accessibilityLabel = Localization.accessibilityLabelCreateCoupons
-        button.accessibilityHint = Localization.accessibilityHintCreateCoupons
-        button.accessibilityIdentifier = "coupon-create-button"
-
-        return button
-    }()
-
     private var subscriptions: Set<AnyCancellable> = []
 
     private lazy var dataSource: UITableViewDiffableDataSource<Section, CouponListViewModel.CellViewModel> = makeDataSource()
     private lazy var topBannerView: TopBannerView = createFeedbackBannerView()
 
-    private lazy var noticePresenter: DefaultNoticePresenter = {
-        let noticePresenter = DefaultNoticePresenter()
-        noticePresenter.presentingViewController = self
-        return noticePresenter
-    }()
+    var onDataLoaded: ((Bool) -> Void)?
+    var noResultConfig: EmptyStateViewController.Config?
+    var onCouponSelected: ((Coupon) -> Void)?
```

Comment:
(❓) Just curious: Since these properties are exposed, is there a reason we're using declaring them as optional and using property injection instead of constructor injection in the `EnhancedCouponListViewController`? This makes the `EnhancedCouponListViewController` having to create its own `CouponListViewController` dependency rather than being passed from outside on class initialization.

Reply:
Good question! I changed this in the next PR. Here I added those as properties to be set instead of being injected in the initialiser because it wasn't possible to pass them to the `CouponListViewController` property  in the EnhancedCouponListViewController `init`, as the `EnhancedCouponListViewController` object is not fully created yet. In the next PR I make the `CouponListViewController` property of `EnhancedCouponListViewController` optional and initialize it after the `init`.